### PR TITLE
mudstepper ux updates

### PR DESCRIPTION
**Pull request by segfault-**
figured i would attempt to contribute.  let me know if there are any issues

### CODE REVIEW

**Comment by AlexandrePViana** [comment]
Maybe it´s just me, I read there was a SetActiveIndex property, and even tho I have 0 idea what the 2 booleans in it mean, I can´t get the MudStepper to go to the step I want it to using code. Plus a Disable feature for the Step would be great, I'm currently using the linear function to try to achieve the same goal, but the disable would be 1000x better so that I could allow users to cycle through the allowed steps before they unlock the next ones, if that makes sense.

**Comment by mckaragoz** [comment]
> Maybe it´s just me, I read there was a SetActiveIndex property, and even tho I have 0 idea what the 2 booleans in it mean, I can´t get the MudStepper to go to the step I want it to using code. Plus a Disable feature for the Step would be great, I'm currently using the linear function to try to achieve the same goal, but the disable would be 1000x better so that I could allow users to cycle through the allowed steps before they unlock the next ones, if that makes sense.  

MudStepper has SetActiveStepByIndex, it's relatively new method to set an active step programmatically

**Comment by mckaragoz** [comment]
It is not looking properly for me, could you check it again?

And it also breaking change for the current design. Example the header doesn't support outlined or text variant.

And doesn't the template increase the complexity? Why don't we change the design directly?

**Comment by segfault-** [comment]
refactored and added vertical support, check again?

**Comment by segfault-** [comment]
as noted in discord, i updated the pr to address the localization issue

File: CodeBeam.MudBlazor.Extensions/Components/Stepper/MudStepTemplate.razor [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿@namespace MudExtensions
+@inherits ComponentBase
+
+@if (Step != null)
+{
+    @if (Step.Status == MudExtensions.Enums.StepStatus.Completed)
+    {
+        <MudBadge Content="3" Overlap="true" Class="" Icon="@Icons.Material.Filled.Done">
```
Comment (mckaragoz) [inline CodeBeam.MudBlazor.Extensions/Components/Stepper/MudStepTemplate.razor:8]: Is there a reason to add `Content="3"` here?

**Comment by mckaragoz** [comment]
@segfault- and do we have breaking change with this PR, we should notice the users.

File: CodeBeam.MudBlazor.Extensions/Components/Stepper/MudStepTemplate.razor (hunk 1 above)
Comment (segfault-) [reply CodeBeam.MudBlazor.Extensions/Components/Stepper/MudStepTemplate.razor:8]: no reason, i must have missed removing that.

**Comment by mckaragoz** [comment]
Thanks 🥇

**Comment by AlexandrePViana** [comment]
Was it just my Stepper that completely changed the visuals? I think either make it possible for fullscreen stepper (no padding on the sides) or have an option to have the previous version available.

**Comment by mckaragoz** [comment]
> Was it just my Stepper that completely changed the visuals? I think either make it possible for fullscreen stepper (no padding on the sides) or have an option to have the previous version available.

Do you have a snippet?

**Comment by AlexandrePViana** [comment]
image
image

Both Images containing the current and previous version. I didn´t make any change to my code. But I leave here the MudStepper Snippet


` <MudStepper @ref="_stepper"  Color="Color.Primary" Variant="Variant.Filled" HeaderTextView="HeaderTextView.All" LocalizedStrings="GetLocalizedStrings()" Loading="_loading"
            PreventStepChangeAsync="new Func<StepChangeDirection, Task<bool>>(CheckChange)" Linear="true" DisableStepResultIndicator="true" DisableNextButton="@(item.idIndex == 2 || item.fechado || forceDisable)" DisablePreviousButton="true" >
<MudStep Title="Criado" Status="@stepStat(0)">
</MudStep>
</MudStepper>`

**Pull request merged**